### PR TITLE
remove ZookeeperInstance and polish

### PR DESCRIPTION
**Pull request by spencergibb**
@marcingrzejszczak what do you think?

### CODE REVIEW

File: spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryClientConfiguration.java [hunk 1]
```diff
@@ -79,6 +76,7 @@ public ZookeeperDiscoveryClient zookeeperDiscoveryClient() {
 		private ZookeeperDependencies zookeeperDependencies;
 
 		@Bean
+		@ConditionalOnMissingBean
```
Comment (marcingrzejszczak) [inline spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryClientConfiguration.java]: Very good idea - IMO we should have as much `@ConditionalOnMissingBean` and `@ConditionalOnProperty` as possible

File: spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/dependency/ZookeeperDependency.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import org.springframework.util.StringUtils;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * @author Spencer Gibb
```
Comment (marcingrzejszczak) [inline spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/dependency/ZookeeperDependency.java]: Can I join the authors? :P

**Comment by marcingrzejszczak** [comment]
Why did you make the code more generic instead of more statically typed?

**Comment by spencergibb** [comment]
So, I don't think we need `ZookeeperInstance`.  I was removing that.  Should I replace the `<?>` with `Object` instead?

**Comment by spencergibb** [comment]
@marcingrzejszczak latest commit fixes the generics?

File: spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryClient.java [hunk 3]
```diff
@@ -50,11 +51,11 @@ public String description() {
 
 	@Override
 	public org.springframework.cloud.client.ServiceInstance getLocalServiceInstance() {
-		ServiceInstance<ZookeeperInstance> serviceInstance = serviceDiscovery.getServiceInstance();
+		ServiceInstance<Object> serviceInstance = serviceDiscovery.getServiceInstance();
```
Comment (marcingrzejszczak) [inline spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryClient.java]: What I was thinking of is that we had

`ServiceInstance<ZookeeperInstance>`

and now we have

`ServiceInstance<Object>`

I'm curious why did we go from `ZookeeperInstance` to `Object`?

File: spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperServiceDiscovery.java [hunk 4]
```diff
@@ -105,16 +105,16 @@ public void build() {
...
 											ApplicationContext context,
 											AtomicInteger port,
 											String host,
 											UriSpec uriSpec) {
 		// @formatter:off
-		serviceInstance.set(ServiceInstance.<ZookeeperInstance>builder()
+		serviceInstance.set(ServiceInstance.builder()
 				.name(appName)
-				.payload(new ZookeeperInstance(context.getId()))
+				//.payload(new Object(context.getId()))
```
Comment (marcingrzejszczak) [inline spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperServiceDiscovery.java]: Why did you comment the payload?

**Comment by marcingrzejszczak** [comment]
BTW what is the reason behind removing the `ZookeeperInstance` ? :P

**Comment by marcingrzejszczak** [comment]
PS Once you merge this code I'll write the missing tests that I wrote about in the morning

**Comment by spencergibb** [comment]
I thought it was to make things simple, don't know where my head was.

**Comment by spencergibb** [comment]
closed via 620b35c7129a85afd51b83295035c82cfd90cb95

**Pull request closed without merging**